Assets/Scripts/NPCs/StateMachine/States/Chase.cs

```diff
@@ -1,103 +1,123 @@
-    {
-        Vector3 targetLocation;
-
-        //The NPC will predict the future position of the player (pursuit behavior)
-        if (_npcManager.GetDistanceWithPlayer() > NPCsGlobalVariables.ChasePlayerRange) // TODO add raycast here
+        public override void Move()
         {
-            ComputeAStar();
-            targetLocation = _currentTargetLocation;
-        }
-        //If the NPC is close to the player, just go directly on him
-        else
-        {
-            _currentTargetLocation = NPCsGlobalVariables.DefaultInitialVector;
-            targetLocation = _playerData.LastSeenPosition;
+            Chasing();
         }
 
-        // TODO Check if we can't see the target (raycast) and do smth
-        Vector3 desiredVelocity = Vector3.Normalize(targetLocation - _npcManager.transform.position) *
-                                  _npcManager.GetModelSpeed(NPCsGlobalVariables.ChaseAcceleration);
+        private void Chasing()
+        {
+            Vector3 velocity = ComputeChaseVelocity();
 
-        Vector3 currentVelocity = _npcManager.Rigidbody.velocity;
+            //The NPC will always looking where it needs to go.
+            float anglePlayerNpc = NpcManager.LookWhereYouAreGoing(velocity);
 
-        Vector3 steering = desiredVelocity - currentVelocity;
+            //If the player is out FOV, NPC stop moving and rotate until the player is back in FOV
+            if (anglePlayerNpc > NpcGlobalVariables.FieldOfView &&
+                NpcManager.GetDistanceWithPlayer() > NpcGlobalVariables.ChasePlayerRange)
+            {
+                velocity = Vector3.zero;
+            }
 
-        Vector3 velocity = currentVelocity + steering;
+            //The NPC always walk forward.
+            NpcManager.SetAnimatorVelocity(Vector3.forward * velocity.magnitude);
+        }
 
-        if (Vector3.Magnitude(velocity) > _npcManager.GetModelSpeed(NPCsGlobalVariables.ChaseMaxVelocity))
+        private Vector3 ComputeChaseVelocity()
         {
-            velocity = (velocity / Vector3.Magnitude(velocity)) *
-                       _npcManager.GetModelSpeed(NPCsGlobalVariables.ChaseMaxVelocity);
-        }
+            
+            ComputeCurrentTargetLocation();
 
-        velocity.y = 0;
+            Vector3 desiredVelocity = Vector3.Normalize(_currentTargetLocation - NpcManager.transform.position) *
+                                      NpcManager.GetModelSpeed(NpcGlobalVariables.ChaseAcceleration);
 
-        return velocity;
-    }
+            Vector3 currentVelocity = NpcManager.Rigidbody.velocity;
 
-    private void ComputeAStar()
-    {
-        Vector3 nodeWithoutY = Vector3.zero;
-        if (_currentTargetLocation != NPCsGlobalVariables.DefaultInitialVector)
-        {
-            nodeWithoutY = new Vector3(_currentTargetLocation.x, 0, _currentTargetLocation.z);
+            Vector3 steering = desiredVelocity - currentVelocity;
+
+            var velocity = ComputeVelocity(currentVelocity, steering);
```

Comment:
Here you substract `desiredVelocity` and `currentVelocity` to pass the result to `ComputeVelocity()` which will add back `currentVelocity`
Why not directly pass `desiredVelocity` to the function ?

Reply:
Good catch! We forgot to take into account acceleration, actually, between the two lines in question